hazelcast/src/main/java/com/hazelcast/config/ConfigXmlGenerator.java

```diff
@@ -204,6 +205,16 @@ private void managementCenterXmlGenerator(XmlGenerator gen, Config config) {
         }
     }
 
+    @PrivateApi
+    public static void listXmlGenerator(XmlGenerator gen, Config config) {
```

Comment:
If I move them to internal package, I still need to make them public. Because I would need to access those methods from `ConfigXmlGenerator`. So:

1. Current: `public` + `@PrivateApi`
2. Review: `public` + `move to the internal package`

If that's the comparison, maybe first option could be preferable since in that case we would have all of the xml generation related methods in one class. Is there a reason to avoid using `@PrivateApi`?

Reply:
We talked about this topic on slack. One good argument for not using `@PrivateApi` is that user need to go to method definition in order to understand whether method is private API or not. With auto completion it's likely that users will miss these methods are private API. I will move these methods to `internal` package.

Reply:
https://github.com/hazelcast/hazelcast/pull/19947/commits/088835e7a07fa549b60c4e74b439aeb45e51588d